logicalplan/distribute.go

```diff
@@ -222,3 +246,51 @@ func isDistributive(expr *parser.Expr) bool {
 
 	return true
 }
+
+// matchesExternalLabels returns false if given matchers are not matching external labels.
+// If true, matchesExternalLabels also returns Prometheus matchers without those matching external labels.
```

Comment:
Where/how does this happen? This only returns `bool` :smile:

Reply:
You're right, I used the store filtering function we have in Thanos as a starting point, but forgot to remove this line.

I don't really see the see the need to override selectors and it also makes the code a bit more complicated.